Monika-After-Story/MonikaModDev
# Dissolve for idle disp and more

I wanted to add dissolves for the idle disp from the get go, but didn't have time for it, so I'm doing it now. With these changes (and ideally with an r7 upgrade) all Monika's sprites will use a nice, smooth transition.

### Changes:
- `Dissolve` transition for Monika's idle displayable. The timings are: `0.25` sec for a pose change, `0.1` sec for an exp change.
- New displayables to handle blinking, winks, and tears. That's because the old ATL were clunky and had numerous bugs. The blinking timings are randomized now.
- `Dissolve` transitions for blinking, winks, and tears. The duration for all of them is `0.1` sec (for many of those we can't use more than that, although we could use different durations if we wanted to, leaving this to the testers).
- Some methods of `MASMoniIdleDisp` got the new `skip_dissolve` parameter. It allows us to skip the transition during the next exp change.
- Test cases for the new displayables.
- A displayable to handle following sprites. Supports right/left variants, has commented code for up/down variants too. Also utilises `Dissolve`s and is more optimized than the current `ConditionSwitch`. NOTE: uses a method from #7006.

### Fixes:
- Fixed a bug with incorrect render size of `MASMonikaRender`.
- Fixed a bug with the broken wink ATL (because I deprecated it lol).
- ~~We don't use `scene` in the `spaceroom` label anymore. RenPy has a bug with images being used in both `old_widget` and `new_widget` of a transform (`Dissolve` in this case). Since `scene` is hiding images, it adds them to `old_widget`, then we're showing them, it adds them to `new_widget`. Because of that the images are getting `st` and `at` for both states at once.~~

### Testing:
- Verify the spaceroom looks good (at startup, during idle, during different transitions).
- Check out the new test cases, you can use them for testing.
- Verify the transitions look good. Not too fast, not too slow, no visual bugs.
- Verify blinking, winks, and streaming tears look good both when used in the idle disp and when `show`en directly.
- Verify the transition from idle to a topic and from a topic back to idle (ideally w/ and w/o forcing an exp at the end) looks good (no seizures or weird movements).
- Verify the top of Monika's head doesn't disappear anymore.
- Verify the following sprites look good. Check the transition from a following to non-following and vice versa.
